vaultclient/iam_auth.go

```diff
@@ -0,0 +1,130 @@
+package vaultclient
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/credentials/ec2rolecreds"
+	"github.com/aws/aws-sdk-go/aws/ec2metadata"
+	"github.com/aws/aws-sdk-go/aws/endpoints"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/sts"
+	"github.com/hashicorp/vault/api"
+	"github.com/hootsuite/vault-ctrl-tool/util"
+	jww "github.com/spf13/jwalterweatherman"
+)
+
+func stsSigningResolver(service, region string, optFns ...func(*endpoints.Options)) (endpoints.ResolvedEndpoint, error) {
```

Comment:
If swaths of this are copied from somewhere, please include a link in a comment for where they came from. It helps troubleshooting in the future if there are unforeseen bugs.